### PR TITLE
Add defensive coding in posts per page sniff

### LINKED ISSUE #2390: Internal error in WordPress/Sniffs/WP/PostsPerPageSniff.php
## Bug Description

This code fails if the `$val` is an empty string.
https://github.com/WordPress/WordPress-Coding-Standards/blob/e02529fec40ff219653182d43866e6d6d9e3b47d/WordPress/Sniffs/WP/PostsPerPageSniff.php#L81
```
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: Uninitialized string offset 0 in /tmp/test/vendor/wp-coding-standards/wpcs/WordPress/Sniffs/WP/PostsPerPageSniff.php on line
   |       | 81 (Internal.Exception)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

## Minimal Code Snippet

The issue happens when running this command:
```bash
vendor/bin/phpcs -s --standard=WordPress --sniffs=WordPress.WP.PostsPerPage test.php
```

... over a file containing this code:
```php
<?php

$query = 'posts_per_page=' . (int) $_POST['limit'];
```

## Error Code

WordPress.WP.PostsPerPage

## Environment

| Question                 | Answer
| ------------------------ | -------
| PHP version              | 8.2.10
| PHP_CodeSniffer version  | 3.7.2
| WordPressCS version      | dev-develop e02529f, also tried with 3.0.1
| PHPCSUtils version       | 1.0.8
| PHPCSExtra version       | 1.1.1
| WordPressCS install type | Composer project local
| IDE (if relevant)        | None

## Tested Against `develop` Branch?
- [x] I have verified the issue still exists in the `develop` branch of WordPressCS.

### PR DESCRIPTION
**Pull request by dingo-d**
This will prevent the undefined offset internal error.

Closes #2390

### CODE REVIEW

**Review by jrfnl — CHANGES_REQUESTED** [review]
@dingo-d Thanks for trying to take this one on.

Unfortunately, this is not the correct fix.

Think of it this way: if the `$val` is an empty string will a numeric comparison ever have any value ? No, it won't.

So instead, this sniff needs to do a check of `$stripped_val` against an empty string, straight after the quotes have been stripped and the function should bow out and return `false` in that case.

The patch also needs a few tests to cover the change - the test from the report + a test with an empty string passed to `posts_per_page` when set as an array.

Probably also a good thing to add a similar test to the `DB/SlowQuery` sniff, which doesn't have the same issue due to the difference in code for `callback`, but still.
```php
$query = 'foo=bar&meta_key=&meta_value=bar';
```

**Comment by dingo-d** [comment]
Fixed the PR according to the comments, much appreciated, should have taken a better look into the sniff.

For the `DB/SlowQuery` sniff I'll check this in a separate PR 👍🏼

**Comment by jrfnl** [comment]
> For the `DB/SlowQuery` sniff I'll check this in a separate PR

The `SlowQuery` sniff only needs the test I provided. The callback doesn't contain logic, so is not susceptible to the same issue. The test would only safeguard that _if logic would be added at some point_, we won't forget to guard against this edge case.

**Review by jrfnl — APPROVED** [review]
LGTM ✅ 

Note: of the tests, only the tests on line 128 and 134 would trigger the notice previously. All the same, the additional tests do test the extra added logic, so 👍🏻 .

Would be nice to see the extra test for `SlowQuery` added as well, but happy to ship this as is.

**Comment by jrfnl** [comment]
@dingo-d P.S.: feel free to squash the commits.

**Review by jrfnl — CHANGES_REQUESTED** [review]
> Do you want me to squash all into one, or separate the test ones from the sniff changes ones?

No, this should be one commit. The tests belong with the functional change, so should be included in the same commit (makes life a lot easier when checking history).

File: WordPress/Sniffs/DB/SlowDBQuerySniff.php [hunk 1]
```diff
@@ -53,6 +54,12 @@ public function getGroups() {
 	 * @return bool Always returns TRUE as the value is irrelevant.
 	 */
 	public function callback( $key, $val, $line, $group ) {
+		$stripped_val = TextStrings::stripQuotes( $val );
+
+		if ( '' === $stripped_val ) {
+			return false;
+		}
+
```
[the remark this answers is not in the document]
Comment (dingo-d) [reply WordPress/Sniffs/DB/SlowDBQuerySniff.php:62]: So just bail out on the empty value?

[the remark this answers is not in the document]
Comment (jrfnl) [reply WordPress/Sniffs/DB/SlowDBQuerySniff.php:62]: I'm not sure how I can make this clearer as I've said this three times now: the _SlowDBQuery sniff_ does not need any changes. All that's needed there is the test.

**Review by GaryJones — APPROVED** [review]
✅

**Review by jrfnl — APPROVED** [review]
